recipes/igblast/meta.yaml

```diff
@@ -1,36 +1,38 @@
+{% set version = "1.15.0" %}
+
 package:
   name: igblast
-  version: "1.10.0"
+  version: "{{ version }}"
 
 about:
   home: http://www.ncbi.nlm.nih.gov/projects/igblast/
   license: Public Domain
```

Comment:
Hm.. given all the libraries that are included, I would be very surprised if the built packages license is still only `Public Domain`.
We should (actually have to) fix this as soon as possible..

Reply:
At least the source tarball carries a bunch of license files:
<details open><summary>

```
$ curl -sL ftp://ftp.ncbi.nih.gov/blast/executables/igblast/release/LATEST/ncbi-igblast-1.15.0-src.tar.gz|tar -tz | grep -i license | sort
```
</summary>

```
ncbi-igblast-1.15.0-src/c++/include/misc/jsonwrapp/rapidjson10/license.txt
ncbi-igblast-1.15.0-src/c++/include/misc/jsonwrapp/rapidjson11/license.txt
ncbi-igblast-1.15.0-src/c++/include/util/bitset/license.txt
ncbi-igblast-1.15.0-src/c++/scripts/projects/blast/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/cobalt/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/corelib/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/datatool/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/dispatcher/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/fwdaemon/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/grid-dashboard/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/grid-python/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/gumbel_params/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/igblast/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/lbsbd/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/magicblast/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/mod_caf/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/mod_loadinfo/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/ncbi_applog/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/ncbi_panfs/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/netcache/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/netschedule/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/netstorage_gc/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/netstorage/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/perl_diag/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/project_tree_builder/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/psg_client/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/public/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/pubseq_gateway/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/python_ncbi_dbapi/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/quickblastp/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/rmblastn/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/test_ccpp_read/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/testres-kernel/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/testres-scheduler/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/testres-testapp/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/testres-testcgi/LICENSE
ncbi-igblast-1.15.0-src/c++/scripts/projects/xmlwrapp/LICENSE
ncbi-igblast-1.15.0-src/c++/src/util/compress/bzip2/LICENSE
ncbi-igblast-1.15.0-src/c++/src/util/lmdb/LICENSE
ncbi-igblast-1.15.0-src/c++/src/util/lmdbxx/UNLICENSE
```
</details>

Reply:
Some indications
<details><summary>

```
$ curl -sL ftp://ftp.ncbi.nih.gov/blast/executables/igblast/release/LATEST/ncbi-igblast-1.15.0-src.tar.gz | tar -xzO | grep -ioE '([^ \-\*#/]+ ){1,4}license' | sort -u
```
</summary>

```
0. This License
24. The license
2. Grant of Copyright License
3. Grant of Patent License
6. Trademarks. This License
a copy of the License
a copy of this License
A copy of this license
(also called "this License
and limitations under the License
any patent license
Apache License
- Apply license
are subject to different license
as modifying the License
available in the file LICENSE
avoid the problematic JSON license
BECAUSE THE LIBRARY IS LICENSE
BitMagic Library License
Boost Software License
both it and this License
case 'L': license
code under the JSON license
compliance with the MIT License
conditions stated in this License
contrib has its own license
copy of the MIT License
copyright license
covered by this License
<dd><p>Display the software version, license
display software version & license
Display the software version, license
distributed under the License
do not accept this License
do not modify the License
d Under the BSD License
d Under the JSON License
d under the MIT License
Each license
echo "Please check license
expressly provided under this License
Foundation may revise this license
from you under this License
General Public License
GNU General Public License
GNU LESSER GENERAL PUBLIC LICENSE
Here is the NCBI license
However, the Lesser license
implemented by public license
in compliance with the License
infringement, then any patent license
instead of to this License
In such case, this License
in the file LICENSE
in this section) patent license
is the original libxmlwrapp license
Lesser General Public License
Library General Public License
Licensed under the Apache License
Licensed under the MIT License
License instead of this License
limitations under the License
<listitem><para>Display the software version, license
made available under the License
not covered by this License
of permissions under this License
of this license
of this License
Open Source Software License
ordinary General Public License
ordinary GNU General Public License
or implied-> See the License
or implied. See the License
Other dependencies and license
other system and a license
Our General Public License
output files to be license
permissions granted by this License
Please read the license
provide additional or different license
Public License
RapidJSON source code is license
recipient automatically receives a license
required to accept this License
restrictive license
result of this License
rights under this License
(See accompanying file LICENSE
See the License
set out in the LICENSE
subsequent revision of the license
terms of any separate license
terms of the BSD license
Terms of the JSON License
Terms of the MIT License
that is granting the License
that refer to this License
the above license
the body of this License
the conditions of this License
the copy of this License
the GNU General Public License
the GNU Library Public License
the "Lesser" General Public License
the Lesser General Public License
the Library under this License
The license
themselves, then this License
The OpenLDAP Public License
the ordinary General Public License
the RapidJSON binary is license
therefore covered by this License
the rest of this License
the scope of this License
the terms of the license
the terms of this License
the work to be license
this Lesser General Public License
this license
this License
This license
this requirement contradicts the license
to apply the Apache License
To apply the Apache License
to You under this License
under terms of this license
under the Boost Software License
under the License
under the OSI-approved BSD License
use specified in this license
version 2.1 of the License
version 2 of the License
version 3 of the License
version number of this License
void license
We call this license
well as the other license
we offer you this license
where such license
whose permissions for other license
will not have their license
with a friendly MIT license
you a royalty free license
your acceptance of this License
zlib under the GNU license
```
</details>

Reply:
IgBLAST is written on top of BLAST, so the problem should be the same in regular BLAST. I think any license concerns should be primarily fixed there, and then one can go back here and do the same thing.

All the LICENSE files in the `c++/scripts/projects/` folders fortunately appear to be false positives, except for ones in the `blast` and `igblast` folders. The other "projects" are not built and/or distributed; apparently whatever method they use to create a source tarball isn’t very precise in what it includes. `blast` and `igblast` are public domain, so we should be good, but the `c++/scripts/projects/igblast/LICENSE` file also contains a copy of the LGPL without any explanation as to why.

I also found [a list of licenses for parts of IgBLAST](https://ncbi.github.io/igblast/dev/exceptions.html). It appears to have been autogenerated.

Reply:
Thanks a bunch for digging into this. I do highly appreciate that!